### PR TITLE
Changes to util needed to merge VSD

### PR DESCRIPTION
**Pull request by martin-cs**
Another PR from the variable-sensitivity-domain branch.  A few minor additions to core things in util/.  I have kept the old commit messages to preserve authorship but TBH they should probably be reworded at least.

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: src/util/symbol_table_base.cpp [hunk 1]
```diff
@@ -60,6 +60,16 @@ void symbol_table_baset::show(std::ostream &out) const
     out << symbols.at(name);
 }
 
+symbol_table_baset::const_iteratort symbol_table_baset::end() const
```
Comment (peterschrammel) [inline src/util/symbol_table_base.cpp:63]: Commit message doesn't match content

File: unit/util/irep.cpp [hunk 2]
```diff
@@ -22,7 +22,7 @@ SCENARIO("irept_memory", "[core][utils][irept]")
       const std::size_t ref_count_size = 0;
 #endif
 
-#ifndef USE_STRING
+#ifndef USE_STD_STRING
```
Comment (peterschrammel) [inline unit/util/irep.cpp:25]: Make commit message title more precise.
"various things": there's only one thing in this commit.

**Comment by martin-cs** [comment]
@peterschrammel Sorry about the commit messages, I kept them from splitting up VSD.  First I wanted to get the code right, then the meta-data.  If you prefer I will squash everything and write meaningful commit messages first.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 5db9d83cb324f199c028b606feef1acf30cf1be7).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/140335402

**Comment by martin-cs** [comment]
@peterschrammel I have re-written the commit messages.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0e3195c5f72307e13fed5439d4d5d11301815630).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/140805372

File: src/util/symbol_table_base.cpp (hunk 1 above)
Comment (hannes-steffenhagen-diffblue) [inline src/util/symbol_table_base.cpp:63]: I’m not sure this change is even necessary; Develop `symbol_table_baset` already seems to have `begin()/end()`.

**Review by hannes-steffenhagen-diffblue — APPROVED** [review]

File: src/util/symbol_table_base.cpp (hunk 1 above)
Comment (thk123) [reply src/util/symbol_table_base.cpp:63]: It has a non-const version - this adds a const version

**Review by thk123 — APPROVED** [review]
Commit messages have been tidied up

**Comment by thk123** [comment]
Brought in via #5375

**Pull request closed without merging**